optaplanner-examples/src/main/java/org/optaplanner/examples/pas/solver/score/PatientAdmissionMoveConstraintProvider.java

```diff
@@ -0,0 +1,182 @@
+import static org.optaplanner.core.api.score.stream.Joiners.greaterThan;
+import static org.optaplanner.core.api.score.stream.Joiners.lessThan;
+
+import java.util.function.Function;
+
+import org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore;
+import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.api.score.stream.ConstraintFactory;
+import org.optaplanner.core.api.score.stream.ConstraintProvider;
+import org.optaplanner.examples.pas.domain.BedDesignation;
+import org.optaplanner.examples.pas.domain.Department;
+import org.optaplanner.examples.pas.domain.DepartmentSpecialism;
+import org.optaplanner.examples.pas.domain.Gender;
+import org.optaplanner.examples.pas.domain.GenderLimitation;
+import org.optaplanner.examples.pas.domain.PreferredPatientEquipment;
+import org.optaplanner.examples.pas.domain.RequiredPatientEquipment;
+import org.optaplanner.examples.pas.domain.RoomEquipment;
+import org.optaplanner.examples.pas.domain.RoomSpecialism;
+
+/*
+ * This is constraints for Hospital Bed Planning
+ * They are based on patientAdmissionScheduleConstraints.drl
+ * Planning Entity: BedDesignation
+ * Planning Variable: Bed(nullable) - would not be prefiltered on uninitialized solutions
+ * Bed is nullable so in case you need to access it members check that planning value bed is not null
+ */
+
+public class PatientAdmissionMoveConstraintProvider implements ConstraintProvider {
+
+    @Override
+    public Constraint[] defineConstraints(ConstraintFactory constraintFactory) {
+        return new Constraint[] {
+                sameBedInSameNightConstraint(constraintFactory),
+                femaleInMaleRoomConstraint(constraintFactory),
+                maleInFemaleRoomConstraint(constraintFactory),
+                differentGenderInSameGenderRoomInSameNightConstraint(constraintFactory),
+                departmentMinimumAgeConstraint(constraintFactory),
+                departmentMaximumAgeConstraint(constraintFactory),
+                requiredPatientEquipmentConstraint(constraintFactory),
+                assignEveryPatientToABedConstraint(constraintFactory),
+                preferredMaximumRoomCapacityConstraint(constraintFactory),
+                departmentSpecialismConstraint(constraintFactory),
+                roomSpecialismNotExistsConstraint(constraintFactory),
+                roomSpecialismNotFirstPriorityConstraint(constraintFactory),
+                preferredPatientEquipmentConstraint(constraintFactory)
+        };
+    }
+
+    public Constraint sameBedInSameNightConstraint(ConstraintFactory constraintFactory) {
+        return constraintFactory.fromUniquePair(BedDesignation.class,
+                        equal(BedDesignation::getBed))
+                .filter((left,right) -> left.getAdmissionPart().calculateSameNightCount(right.getAdmissionPart()) > 0
+                && left.getBed()!=null && right.getBed()!=null)
```

Comment:
Please put the `null` checks first in the filter.
Also, since `BedDesignation::getBed` is `equal`, you only need to check if `left.getBed() != null`.

Reply:
Yeh, the last one I got a call from somewhere inside, thanks for correcting.